httpbakery/client.go

```diff
@@ -428,9 +428,10 @@ func (c *Client) interact(location, visitURLStr, waitURLStr string) (*macaroon.M
 		if err := json.NewDecoder(waitResp.Body).Decode(&resp); err != nil {
 			return nil, errgo.Notef(err, "cannot unmarshal wait error response")
 		}
-		return nil, errgo.WithCausef(nil, &DischargeError{
+		dischargeErr := &DischargeError{
```

Comment:
Yes, I plan to do that in a future PR, probably by implementing it in httprequest and using that in httpbakery.